### DISCUSSION TITLE
采集java异常堆栈日志时，没有按示例合并为一条

### DISCUSSION THREAD

**Discussion by xiaotangyuan0106** [alibaba/ilogtail#1706] [Help] [answered]
ilogtail版本：2.0.0
采集配置如下：
![image](https://github.com/user-attachments/assets/1868c427-8595-404f-9a22-4ef80d5cf1b5)
ilogtail的主程序运行日志有error错误日志，如下：
![image](https://github.com/user-attachments/assets/6f52c17a-5fe8-41d3-9780-a6eecb7111e5)
正则我也验过了，没有问题。
业务日志如下：
![image](https://github.com/user-attachments/assets/6c4b5112-4ef4-4148-aca7-2167312844d1)
不知道哪里配置的有问题，大佬们，有空帮忙看一眼，谢谢了

**Comment by Takuka0311** [comment, marked as answer]
StartPattern需要匹配完整的第一行，不是第一行的开头

**Reply by xiaotangyuan0106** [reply]
好的，谢谢大佬，改为匹配完整第一行，可以了。

**Reply by Takuka0311** [reply]
好的。可以在这里登记下您的使用场景：https://github.com/alibaba/ilogtail/issues/693
